core/client/src/main/java/alluxio/client/block/UnderStoreBlockInStream.java

```diff
@@ -52,17 +53,18 @@
    */
   private long mPos;
   /** The current under store stream. */
-  private InputStream mUnderStoreStream;
+  private UnderFileInputStream mUnderStoreStream;
 
   /**
    * A factory which can create an input stream to under storage.
    */
   public interface UnderStoreStreamFactory extends AutoCloseable {
     /**
+     * @param options for opening a UFS input stream
      * @return an input stream to under storage
      * @throws IOException if an IO exception occurs
      */
-    InputStream create() throws IOException;
+    UnderFileInputStream create(OpenOptions options) throws IOException;
```

Comment:
Do you think you can skip creating this new type because i will have to change it to InputStream when I merge your PR to mine (https://github.com/Alluxio/alluxio/pull/4427) as I'll provide a different implementation of the UnderFileSystemInStream and i won't be able to extend from the type (UnderFileInputStream) you have added here. 

You can still use InputStream and cast it to Seekable when you need to call seek.  Thank you.  

How do you think?